common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -405,19 +469,35 @@ protected boolean runAllTasks(long timeoutNanos) {
             if ((runTasks & 0x3F) == 0) {
                 lastExecutionTime = ScheduledFutureTask.nanoTime();
                 if (lastExecutionTime >= deadline) {
+                    result = DeadlineRunResult.TIMED_OUT;
```

Comment:
I think the return status and the ability to get the last execution time are two different concerns. eg: The only reason `SingleThreadEventLoop` is using this callback is to avoid doing `System.nanoTime` call before and after `runAllTasksFrom`. This can probably be questioned BUT the case in point being that the callback is _optional_. I can imagine a caller which is just interested in the status of executing these tasks as opposed to finding the times, in which case using a callback for getting the third status is awkward.